test/comps/test_orth.cc

```diff
@@ -24,41 +24,52 @@ class TestOrth : public ::testing::Test
 
     virtual void TearDown() {};
 
-    /// Tests orthogonality of a matrix Q, obtained by orthogonalizing a Gaussian sketch.
-    /// Checks I - \transpose{Q}Q.
-    template <typename T, typename RNG>
-    static void test_orth_sketch(int64_t m, int64_t n, int64_t k, std::tuple<int, T, bool> mat_type, RandBLAS::base::RNGState<RNG> state) {
-
-        int64_t size = m * n;
-        std::vector<T> A(size, 0.0);
-        std::vector<T> Y(m * k, 0.0);
-        std::vector<T> Omega(n * k, 0.0);
-        std::vector<T> I_ref(k * k, 0.0);
-
-        T* A_dat = A.data();
-        T* Y_dat = Y.data();
-        T* Omega_dat = Omega.data();
-        T* I_ref_dat = I_ref.data();
-
-        RandLAPACK::util::gen_mat_type(m, n, A, k, state, mat_type);
+    template <typename T>
+    struct OrthTestData {
+        std::vector<T> A;
+        std::vector<T> Y;
+        std::vector<T> Omega;
+        std::vector<T> I_ref;
+
+        OrthTestData(int64_t m, int64_t n, int64_t k) :
+        A(m * n, 0.0),
+        Y(m * k, 0.0),
+        Omega(n * k, 0.0),
+        I_ref(k * k, 0.0)
+        {}
+    };
 
+    template <typename T, typename RNG>
+    static void computational_helper(int64_t m, int64_t n, int64_t k, RandBLAS::base::RNGState<RNG> state, OrthTestData<T>& all_data) {
         // Fill the gaussian random matrix
         RandBLAS::dense::DenseDist D{.n_rows = n, .n_cols = k};
-        state = RandBLAS::dense::fill_buff(Omega_dat, D, state);
+        state = RandBLAS::dense::fill_buff(all_data.Omega.data(), D, state);
+        
         // Generate a reference identity
-        RandLAPACK::util::eye(k, k, I_ref);
+        RandLAPACK::util::eye(k, k, all_data.I_ref);
+
         // Y = A * Omega
-        blas::gemm(Layout::ColMajor, Op::NoTrans, Op::NoTrans, m, k, n, 1.0, A_dat, m, Omega_dat, n, 0.0, Y_dat, m);
+        blas::gemm(Layout::ColMajor, Op::NoTrans, Op::NoTrans, m, k, n, 1.0, all_data.A.data(), m, all_data.Omega.data(), n, 0.0, all_data.Y.data(), m);
+    }
```

Comment:
Give this function a more informative name.

Reply:
I kept that name across all file so that the reader can get a feeling for the fact that the testing structure is universal.

Reply:
Fixed.

Reply:
Fixed.